app/models/solr_document.rb

```diff
@@ -1,3 +1,112 @@
-class SolrDocument 
+class SolrDocument
+
   include Blacklight::Solr::Document
+
+  attr_accessor :json
+
+  def json
+    @json ||= (JSON.parse(@_source[:json]) rescue {}).with_indifferent_access
+  end
+
+  def label_for(field)
+    # Assume labels are the same as field names, sans underscores and
+    # "clip_" prefix if it has one.
+    field.to_s.gsub('_',' ').gsub(/clip /i, '')
+  end
+
+  def labels_and_values(*fields)
+    fields.map{ |field| { label: label_for(field), value: send(field) } }.select{ |pair| !pair[:value].nil? && !pair[:value].empty? }
+  end
```

Comment:
`label_for` could be a class method, but values come from an instance. And speaking of field labels, I'd like to move in the direction of using `I18n` where applicable.